### PR TITLE
refactor: cli/js

### PR DESCRIPTION
**Pull request by stanislavstrelnikov**

### CODE REVIEW

File: cli/js/web/url.ts [hunk 1]
```diff
@@ -408,7 +409,7 @@ export class URLImpl implements URL {
 
   constructor(url: string | URL, base?: string | URL) {
     let baseParts: URLParts | undefined;
-    if (base) {
+    if (base != null) {
```
Comment (marcosc90) [inline cli/js/web/url.ts:412]: What's the reason behind these changes? I for one prefer the current way, the intent is clear, you expect a _truthy_ value, on the other hand when using `!= null` it can either be a bug (=== should be used) or you expect a truthy value. 

Is there a style guide?

File: cli/js/web/blob.ts [hunk 2]
```diff
@@ -6,11 +6,11 @@ import { ReadableStreamImpl } from "./streams/readable_stream.ts";
 
 export const bytesSymbol = Symbol("bytes");
 
-export function containsOnlyASCII(str: string): boolean {
-  if (typeof str !== "string") {
+export function containsOnlyASCII(s: string): boolean {
```
Comment (marcosc90) [inline cli/js/web/blob.ts:9]: While I agree with many of the changes, a lot of them seem arbitrary and based on your personal preference, is there a code style that we're following that justify these changes?

Comment (stanislavstrelnikov) [reply cli/js/web/blob.ts:9]: > While I agree with many of the changes, a lot of them seem arbitrary and based on your personal preference, is there a code style that we're following that justify these changes?

In this case, i just wanted parameter name to match other functions in this file

File: cli/js/web/abort_controller.ts [hunk 3]
```diff
@@ -3,14 +3,10 @@
...
-  #signal = new AbortSignalImpl();
-
-  get signal(): AbortSignal {
-    return this.#signal;
-  }
+  readonly signal = new AbortSignalImpl();
 
   abort(): void {
-    this.#signal[signalAbort]();
+    this.signal[signalAbort]();
```
Comment (marcosc90) [inline cli/js/web/abort_controller.ts:9]: This changes the behavior, now it's possible to do:

```
const signal = new AbortController();
signal.signal = 5;
signal.signal === 5; // true
```

File: cli/js/tls.ts [hunk 4]
```diff
@@ -66,7 +66,7 @@ interface StartTlsOptions {
 
 export async function startTls(
   conn: Conn,
-  { hostname = "127.0.0.1", certFile }: StartTlsOptions = {}
+  { hostname = "127.0.0.1", certFile }: StartTlsOptions
```
Comment (nayeemrmn) [inline cli/js/tls.ts:69]: This is incorrect.

File: cli/js/web/blob.ts [hunk 5]
```diff
@@ -173,7 +173,8 @@ class DenoBlob implements Blob {
       return;
     }
 
-    const { ending = "transparent", type = "" } = options ?? {};
+    const ending = options?.ending ?? "transparent";
+    const type = options?.type ?? "";
```
Comment (nayeemrmn) [inline cli/js/web/blob.ts:177]: The original style is preferred.

File: cli/js/web/event_target.ts [hunk 6]
```diff
@@ -292,7 +292,7 @@ function innerInvokeEventListeners(
 ): boolean {
   let found = false;
 
-  const { type } = eventImpl;
+  const type = eventImpl.type;
```
Comment (nayeemrmn) [inline cli/js/web/event_target.ts:295]: See earlier comment.

File: cli/js/web/event_target.ts [hunk 7]
```diff
@@ -498,7 +498,7 @@ export class EventTargetImpl implements EventTarget {
     }
 
     options = normalizeAddEventHandlerOptions(options);
-    const { listeners } = eventTargetData.get(this ?? globalThis)!;
+    const listeners = eventTargetData.get(this ?? globalThis)!.listeners;
```
Comment (nayeemrmn) [inline cli/js/web/event_target.ts:501]: See earlier comment.

File: cli/js/web/fetch.ts [hunk 8]
```diff
@@ -33,7 +33,8 @@ export class Response extends Body.Body implements domTypes.Response {
     }
 
     const extraInit = responseData.get(init) || {};
-    let { type = "default", url = "" } = extraInit;
+    let type = extraInit.type ?? "default";
+    let url = extraInit.url ?? "";
```
Comment (nayeemrmn) [inline cli/js/web/fetch.ts:37]: See earlier comment.

File: cli/js/web/streams/internals.ts [hunk 9]
```diff
@@ -388,7 +388,7 @@ export function isTransformStreamDefaultController(
 export function isUnderlyingByteSource(
   underlyingSource: UnderlyingByteSource | UnderlyingSource
 ): underlyingSource is UnderlyingByteSource {
-  const { type } = underlyingSource;
+  const type = underlyingSource.type;
```
Comment (nayeemrmn) [inline cli/js/web/streams/internals.ts:391]: See earlier comment.

File: cli/js/web/streams/internals.ts [hunk 10]
```diff
@@ -1218,7 +1211,7 @@ export function readableStreamTee<T>(
       (result) => {
         reading = false;
         assert(typeof result === "object");
-        const { done } = result;
+        const done = result.done;
```
Comment (nayeemrmn) [inline cli/js/web/streams/internals.ts:1214]: See earlier comment.

File: cli/js/web/streams/internals.ts [hunk 11]
```diff
@@ -1237,7 +1230,7 @@ export function readableStreamTee<T>(
           }
           return;
         }
-        const { value } = result;
+        const value = result.value;
```
Comment (nayeemrmn) [inline cli/js/web/streams/internals.ts:1233]: See earlier comment.

File: cli/js/web/streams/queuing_strategy.ts [hunk 12]
```diff
@@ -3,11 +3,15 @@
...
+  highWaterMark: number;
+}
+
 export class CountQueuingStrategyImpl implements CountQueuingStrategy {
   highWaterMark: number;
 
-  constructor({ highWaterMark }: { highWaterMark: number }) {
-    this.highWaterMark = highWaterMark;
+  constructor(options: QueuingStrategyOptions) {
+    this.highWaterMark = options.highWaterMark;
```
Comment (nayeemrmn) [inline cli/js/web/streams/queuing_strategy.ts:14]: The original style is preferred.

File: cli/js/web/streams/queuing_strategy.ts [hunk 13]
```diff
@@ -31,8 +35,8 @@ export class ByteLengthQueuingStrategyImpl
   implements ByteLengthQueuingStrategy {
   highWaterMark: number;
 
-  constructor({ highWaterMark }: { highWaterMark: number }) {
-    this.highWaterMark = highWaterMark;
+  constructor(options: QueuingStrategyOptions) {
+    this.highWaterMark = options.highWaterMark;
```
Comment (nayeemrmn) [inline cli/js/web/streams/queuing_strategy.ts:39]: See earlier comment.

File: cli/js/web/streams/readable_stream.ts [hunk 14]
```diff
@@ -47,9 +55,9 @@ export class ReadableStreamImpl<R = any> implements ReadableStream<R> {
       | QueuingStrategy<R> = {}
   ) {
     initializeReadableStream(this);
-    const { size } = strategy;
-    let { highWaterMark } = strategy;
-    const { type } = underlyingSource;
+    const size = strategy.size;
+    let highWaterMark = strategy.highWaterMark;
+    const type = underlyingSource.type;
```
Comment (nayeemrmn) [inline cli/js/web/streams/readable_stream.ts:60]: See earlier comment.

File: cli/js/web/streams/writable_stream.ts [hunk 15]
```diff
@@ -41,7 +41,7 @@ export class WritableStreamImpl<W = any> implements WritableStream<W> {
     initializeWritableStream(this);
     const size = strategy.size;
     let highWaterMark = strategy.highWaterMark ?? 1;
-    const { type } = underlyingSink;
+    const type = underlyingSink.type;
```
Comment (nayeemrmn) [inline cli/js/web/streams/writable_stream.ts:44]: See earlier comment.

File: cli/js/web/timers.ts [hunk 16]
```diff
@@ -4,7 +4,7 @@ import { assert } from "../util.ts";
 import { startGlobalTimer, stopGlobalTimer } from "../ops/timers.ts";
 import { RBTree } from "../rbtree.ts";
 
-const { console } = globalThis;
+const console = globalThis.console;
```
Comment (nayeemrmn) [inline cli/js/web/timers.ts:7]: See earlier comment.

File: cli/js/web/url.ts [hunk 17]
```diff
@@ -135,8 +136,8 @@ function isAbsolutePath(path: string): boolean {
 // Preserves repeating and trailing `/`s by design.
 // On Windows, drive letter paths will be given a leading slash, and also a
 // trailing slash if there are no other components e.g. "C:" -> "/C:/".
-function normalizePath(path: string, isFilePath = false): string {
-  if (build.os == "windows" && isFilePath) {
+function normalizePath(path: string, filePath = false): string {
+  if (build.os == "windows" && filePath) {
```
Comment (nayeemrmn) [inline cli/js/web/url.ts:140]: The original name is better.

File: cli/js/web/url.ts [hunk 18]
```diff
@@ -163,7 +164,7 @@ function normalizePath(path: string, isFilePath = false): string {
       newPath = ".";
     }
   } else {
-    newPath = `/${newPath}`;
+    newPath = "/" + newPath;
```
Comment (nayeemrmn) [inline cli/js/web/url.ts:167]: See earlier comment.

File: cli/js/web/url.ts [hunk 19]
```diff
@@ -388,7 +389,7 @@ export class URLImpl implements URL {
 
   set search(value: string) {
     value = String(value);
-    const query = value == "" || value.charAt(0) == "?" ? value : `?${value}`;
+    const query = value === "" || value.charAt(0) === "?" ? value : "?" + value;
```
Comment (nayeemrmn) [inline cli/js/web/url.ts:392]: See earlier comment.

File: cli/js/web/workers.ts [hunk 20]
```diff
@@ -84,21 +82,22 @@ export class WorkerImpl extends EventTarget implements Worker {
...
+    const type = options?.type ?? "classic";
     if (type !== "module") {
       throw new Error(
         'Not yet implemented: only "module" type workers are supported'
       );
     }
 
-    this.#name = name;
+    this.#name = options?.name ?? "unknown";
+
```
Comment (nayeemrmn) [inline cli/js/web/workers.ts:100]: See earlier comment. The defaults should be written together higher up.

File: cli/js/web/abort_controller.ts (hunk 3 above)
Comment (kitsonk) [reply cli/js/web/abort_controller.ts:9]: Yes, this is against the spec, why make this change?

File: cli/js/compiler.ts [hunk 21]
```diff
@@ -523,7 +523,7 @@ class IncrementalCompileHost extends Host {
 
   constructor(options: IncrementalCompilerHostOptions) {
     super(options);
-    const { buildInfo } = options;
+    const buildInfo = options.buildInfo;
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply cli/js/compiler.ts:526]: Correct.

File: cli/js/compiler.ts [hunk 22]
```diff
@@ -979,7 +979,8 @@ function processConfigureResponse(
   configResult: ConfigureResponse,
   configPath: string
 ): ts.Diagnostic[] | undefined {
-  const { ignoredOptions, diagnostics } = configResult;
+  const ignoredOptions = configResult.ignoredOptions;
+  const diagnostics = configResult.diagnostics;
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply cli/js/compiler.ts:983]: Correct, why are you trying to change these‽

File: cli/js/build.ts [hunk 23]
```diff
@@ -1,11 +1,19 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
-export const build = {
+interface BuildInfo {
```
Comment (kitsonk) [inline cli/js/build.ts:3]: Why add an interface here, it doesn't add any value.

File: cli/js/web/abort_controller.ts (hunk 3 above)
Comment (stanislavstrelnikov) [reply cli/js/web/abort_controller.ts:9]: > Yes, this is against the spec, why make this change?

My bad, i'll revert it

File: cli/js/colors.ts [hunk 24]
```diff
@@ -15,9 +15,9 @@ function code(open: number, close: number): Code {
 }
 
 function run(str: string, code: Code): string {
-  return !globalThis || !globalThis.Deno || globalThis.Deno.noColor
-    ? str
-    : `${code.open}${str.replace(code.regexp, code.open)}${code.close}`;
+  return !globalThis.Deno.noColor
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply cli/js/colors.ts:18]: Correct.  This will now throw in certain cases.  Please revert.

File: cli/js/compiler_api.ts [hunk 25]
```diff
@@ -12,7 +12,7 @@ import { CompilerOptions } from "./compiler_options.ts";
 function checkRelative(specifier: string): string {
   return specifier.match(/^([\.\/\\]|https?:\/{2}|file:\/{2})/)
     ? specifier
-    : `./${specifier}`;
+    : "./" + specifier;
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply cli/js/compiler_api.ts:15]: Correct.

File: cli/js/io.ts [hunk 26]
```diff
@@ -50,12 +50,18 @@ export interface SeekerSync {
...
 }
 
+interface CopyOptions {
+  bufSize?: number;
+}
+
+interface IterOptions {
+  bufSize?: number;
+}
+
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply cli/js/io.ts:60]: Agreed.

File: cli/js/ops/fs_events.ts [hunk 27]
```diff
@@ -16,8 +16,10 @@ class FsWatcher implements AsyncIterableIterator<FsEvent> {
   readonly rid: number;
 
   constructor(paths: string[], options: FsWatcherOptions) {
-    const { recursive } = options;
-    this.rid = sendSync("op_fs_events_open", { recursive, paths });
+    this.rid = sendSync("op_fs_events_open", {
```
Comment (kitsonk) [inline cli/js/ops/fs_events.ts:19]: The original style is preferred, please revert.

File: cli/js/process.ts [hunk 28]
```diff
@@ -48,11 +48,11 @@ export class Process<T extends RunOptions = RunOptions> {
     this.rid = res.rid;
     this.pid = res.pid;
 
-    if (res.stdinRid && res.stdinRid > 0) {
+    if (res.stdinRid !== null && res.stdinRid > 0) {
       this.stdin = (new File(res.stdinRid) as unknown) as Process<T>["stdin"];
     }
 
-    if (res.stdoutRid && res.stdoutRid > 0) {
+    if (res.stdoutRid !== null && res.stdoutRid > 0) {
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply cli/js/process.ts:55]: Agreed.  Though in this case the original code was perfectly fine, because we don't want 0 anyways.

File: cli/js/read_text_file.ts [hunk 29]
```diff
@@ -7,14 +7,12 @@ export function readTextFileSync(path: string | URL): string {
   const file = openSync(path);
   const contents = readAllSync(file);
   file.close();
-  const decoder = new TextDecoder();
-  return decoder.decode(contents);
+  return new TextDecoder().decode(contents);
```
Comment (kitsonk) [inline cli/js/read_text_file.ts:10]: In fact, the module should declare a `const decoder = new TextDecoder();` and be used here and the next function instead of creating a new decoder each call.

File: cli/js/tls.ts [hunk 30]
```diff
@@ -44,10 +44,10 @@ export interface ListenTlsOptions {
 
 export function listenTls({
   port,
-  certFile,
-  keyFile,
   hostname = "0.0.0.0",
   transport = "tcp",
+  certFile,
```
Comment (kitsonk) [inline cli/js/tls.ts:49]: If we are going to move them about, we should alphabetise them.

File: cli/js/tls.ts (hunk 4 above)
Comment (kitsonk) [reply cli/js/tls.ts:69]: Yes. Please revert.

File: cli/js/util.ts [hunk 31]
```diff
@@ -53,9 +51,9 @@ export type Resolvable<T> = Promise<T> & ResolvableMethods<T>;
 export function createResolvable<T>(): Resolvable<T> {
   let resolve: ResolveFunction<T>;
   let reject: RejectFunction;
-  const promise = new Promise<T>((res, rej): void => {
-    resolve = res;
-    reject = rej;
+  const promise = new Promise<T>((_resolve, _reject): void => {
```
Comment (kitsonk) [inline cli/js/util.ts:54]: The previous style is preferred, please revert.

File: cli/js/util.ts [hunk 32]
```diff
@@ -110,17 +108,17 @@ function pathFromURLPosix(url: URL): string {
   return decodeURIComponent(url.pathname);
 }
 
-export function pathFromURL(pathOrUrl: string | URL): string {
-  if (pathOrUrl instanceof URL) {
-    if (pathOrUrl.protocol != "file:") {
+export function pathFromURL(url: string | URL): string {
```
Comment (kitsonk) [inline cli/js/util.ts:111]: The previous style is preferred, please revert.

File: cli/js/web/abort_controller.ts [hunk 33]
```diff
@@ -3,14 +3,10 @@
 import { AbortSignalImpl, signalAbort } from "./abort_signal.ts";
 
 export class AbortControllerImpl implements AbortController {
-  #signal = new AbortSignalImpl();
-
-  get signal(): AbortSignal {
-    return this.#signal;
-  }
+  readonly signal = new AbortSignalImpl();
```
Comment (kitsonk) [inline cli/js/web/abort_controller.ts:6]: This is wrong.  `readonly` is a TypeScript only constraint.  We need real runtime read-only to be spec compliant.  Please revert.

File: cli/js/web/abort_signal.ts [hunk 34]
```diff
@@ -35,10 +38,10 @@ export class AbortSignalImpl extends EventTargetImpl implements AbortSignal {
 
   constructor() {
     super();
+
     this.addEventListener("abort", (evt: Event) => {
-      const { onabort } = this;
-      if (typeof onabort === "function") {
-        onabort.call(this, evt);
+      if (typeof this.onabort === "function") {
```
Comment (kitsonk) [inline cli/js/web/abort_signal.ts:43]: The previous style is preferred, please revert.

File: cli/js/web/body.ts [hunk 35]
```diff
@@ -157,44 +150,36 @@ export class Body implements domTypes.Body {
...
+      return this._bodySource;
     } else if (typeof this._bodySource === "string") {
-      const enc = new TextEncoder();
-      return Promise.resolve(
-        enc.encode(this._bodySource).buffer as ArrayBuffer
-      );
+      return encoder.encode(this._bodySource).buffer;
     } else if (this._bodySource instanceof ReadableStreamImpl) {
-      return bufferFromStream(this._bodySource.getReader());
+      const buf = await bufferFromStream(this._bodySource.getReader());
```
Comment (kitsonk) [inline cli/js/web/body.ts:174]: This await is totally unnecessary.  Just return the value.  It causes an extra promise to be created. 😢

File: cli/js/web/custom_event.ts [hunk 36]
```diff
@@ -5,18 +5,12 @@ import { requiredArguments } from "./util.ts";
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export class CustomEventImpl<T = any> extends Event implements CustomEvent {
-  readonly #detail: T;
+  readonly detail: T;
```
Comment (kitsonk) [inline cli/js/web/custom_event.ts:8]: Please revert.  `readonly` is a TypeScript only construct and to be spec compliant it needs to be truly read-only.

File: cli/js/web/dom_exception.ts [hunk 37]
```diff
@@ -1,14 +1,7 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
 export class DOMExceptionImpl extends Error implements DOMException {
-  readonly #name: string;
```
Comment (kitsonk) [inline cli/js/web/dom_exception.ts:4]: Again, please revert!

**Comment by caspervonb** [comment]
I'm going to play devils advocate here for a minute.

I'm not really comfortable with a bunch of "lots a changes" commits landing with no defined scope or intent spanning across the entire cli code base.

While some of these are applicable and make the code somewhat clearer, some are breaking changes and a ton are just very minor stylistic nits.

It's inevitable that we'll miss something and reviewing "batch" commits like this is a bit of a headache.

**Comment by stanislavstrelnikov** [comment]
> I'm going to play devils advocate here for a minute.
> 
> I'm not really comfortable with a bunch of "lots a changes" commits landing with no defined scope or intent spanning across the entire cli code base.
> 
> While some of these are applicable and make the code somewhat clearer, some are breaking changes and a ton are just very minor stylistic nits.
> 
> It's inevitable that we'll miss something and reviewing "batch" commits like these are a bit of a headache.

You right, it is my bad at some point
